libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -23,9 +23,9 @@
 #define MQTT_LIGHTWEIGHT_H
 
 /* bools are only defined in C99+ */
-#if __STDC_VERSION__ >= 199901L
+#if defined( __cplusplus ) || __STDC_VERSION__ >= 199901L
```

Comment:
(Probably out-of-scope of the PR). Do we want to keep such a check in the code? I thought we are heading to keep no references of C99 in the code?

Reply:
We either need the check or change all the booleans to some other macro or enum. There hasn't been any indication that we don't want to use the `bool` macro, so this defines it for C90.

Reply:
> change all the booleans to some other macro or enum. 

May be better to do.